include/jemalloc/internal/tcache_externs.h

```diff
@@ -71,6 +71,9 @@ void tcache_postfork_parent(tsdn_t *tsdn);
 void tcache_postfork_child(tsdn_t *tsdn);
 void tcache_flush(tsd_t *tsd);
 bool tsd_tcache_data_init(tsd_t *tsd, arena_t *arena);
+bool tsd_tcache_data_init(tsd_t *tsd, arena_t *arena);
+bool tsd_tcache_data_init_with_bin_info(tsd_t *tsd, arena_t *arena,
```

Comment:
Doesn't appear to be used

Reply:
Now used in `thread_tcache_max_set`

Reply:
Now tsd_tcache_data_init is not used. Also there's an duplicated line above.

We should keep only tsd_tcache_data_init, which takes the new bin_info arg.

Reply:
`tsd_tcache_data_init_with_bin_info` is used to init a tcache with the previous ncached_max carried over. The inits before the cleanup is to save the bin_info. The name `cache_bin_info_init` is indeed a bit confusing there, but it's simply assigning the value to a local array `tcache_bin_info`.

Reply:
I see. Let's abstract that for loop out, and name it to say what it's doing, e.g. `tcache_settings_backup`. Then rename tsd_tcache_data_init_with_bin_info to something symmetric, e.g. `tcache_data_init_with_settings`